count_cells.py

```diff
@@ -39,87 +43,87 @@ def __init__(self, argv):
         self.args = parser.parse_args(argv)
 
     def run(self):
-
         if self.args.verbose:
             logging.getLogger().setLevel(logging.DEBUG)
 
-        project_dirs = get_target_project_dirs()
+        if self.args.accession:
+            accessions = [self.args.accession]
+        else:
+            accessions = [p.name for p in get_target_project_dirs()]
 
-        for project_dir in project_dirs:
-            self.count_one_project_cells(project_dir.name)
+        for accession in accessions:
+            cell_count = self.get_project_cell_count(accession)
+            if self.args.write:
+                self.write_cell_count(accession, cell_count)
 
-    def count_one_project_cells(self, accession_id: str):
+    @classmethod
+    def get_cached_cell_count(cls, accession: str) -> int:
         """
-        Count cells in one project
-
-        :param accession_id: An accession id with a downloaded matrix file
-        :param save_to_file: If true write the cell count total(s) to a global cell count file
+        Return a cell count for the given accession
         """
-        cell_count = self.get_project_cell_count(accession_id)
-        if self.args.write:
-            self.update_cell_count_file(accession_id, cell_count)
+        stats_file = Path(f'projects/{accession}/stats.json')
+        if stats_file.exists():
+            with open(stats_file, 'r') as f:
+                cell_count = json.loads(f.read()).get('cell_count', 0)
+            return cell_count
+        else:
+            return 0
 
     @classmethod
-    def get_accession_ids(cls) -> Sequence[str]:
+    def get_cached_cell_counts(cls) -> Mapping[str, int]:
```

Comment:
```suggestion
    def get_cached_cell_counts(cls) -> MutableMapping[str, int]:
```

Rule of thumb for type hints: be as general as possible for arguments and be as specific as possible for return types. If it's OK for the caller to modify a return value, the callee returning it should explicitly declare that.